From 01b4475b60521b46489593130044b89e8d985770 Mon Sep 17 00:00:00 2001
From: Olivier Naudan <o-naudan@ti.com>
Date: Mon, 21 Mar 2011 16:16:08 +0100
Subject: [PATCH] Include path is incorrect for Tiler header files

---
 phase1_d2c_remap.c |   12 ++++++------
 tests/d2c_test.c   |   10 +++++-----
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/phase1_d2c_remap.c b/phase1_d2c_remap.c
index 468a3d4..86f5c07 100644
--- a/phase1_d2c_remap.c
+++ b/phase1_d2c_remap.c
@@ -34,14 +34,14 @@
 #ifdef HAVE_CONFIG_H
     #include "config.h"
 #endif
-#include "mem_types.h"
 #include "utils.h"
 #include "debug_utils.h"
-#include "list_utils.h"
-#include "tilermem_utils.h"
+#include "testlib.h"
 #include "phase1_d2c_remap.h"
-#include "memmgr.h"
-#include "tiler.h"
+#include <timemmgr/memmgr.h>
+#include <timemmgr/tilermem_utils.h>
+#include <timemmgr/tilermem.h>
+#include <timemmgr/tiler.h>
 
 /* ----- START debug only methods ----- */
 
@@ -426,4 +426,4 @@ int tiler_assisted_phase1_DeMap(void *bufPtr)
     close(td);
 #endif
     return R_I(ret);
-}
\ No newline at end of file
+}
diff --git a/tests/d2c_test.c b/tests/d2c_test.c
index 165c932..5479859 100644
--- a/tests/d2c_test.c
+++ b/tests/d2c_test.c
@@ -31,14 +31,14 @@
 #ifdef HAVE_CONFIG_H
     #include "config.h"
 #endif
-#include <memmgr.h>
-#include <phase1_d2c_remap.h>
 #include <utils.h>
 #include <debug_utils.h>
-#include <tilermem_utils.h>
-#include <tilermem.h>
-#include <tiler.h>
 #include <testlib.h>
+#include <phase1_d2c_remap.h>
+#include <timemmgr/memmgr.h>
+#include <timemmgr/tilermem_utils.h>
+#include <timemmgr/tilermem.h>
+#include <timemmgr/tiler.h>
 
 #define TESTS\
     T(test_d2c(176, 144, 1))\
-- 
1.7.1

