Description: Make compile with latest GIR
Author: Robert Ancell

Index: avahi-0.6.27/avahi-gobject/AvahiCore-0.6.gir
===================================================================
--- avahi-0.6.27.orig/avahi-gobject/AvahiCore-0.6.gir	2010-08-25 15:16:54.395928001 +1000
+++ avahi-0.6.27/avahi-gobject/AvahiCore-0.6.gir	2010-08-25 15:17:02.805928001 +1000
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<repository version="1.0"
+<repository version="1.1"
             xmlns="http://www.gtk.org/introspection/core/1.0"
             xmlns:c="http://www.gtk.org/introspection/c/1.0"
             xmlns:glib="http://www.gtk.org/introspection/glib/1.0">
