From 77004307ba2a677d394ec52ea78347e3e4bd2e86 Mon Sep 17 00:00:00 2001
From: Reynaldo H. Verdejo Pinochet <reynaldo.verdejo@collabora.com>
Date: Sun, 29 May 2011 17:53:35 -0400
Subject: [PATCH 7/7] Resume ports in ready to paused to avoid issue in play-stop-play transition

---
 omx/gstomx_base_filter.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/omx/gstomx_base_filter.c b/omx/gstomx_base_filter.c
index 17bc4a9..8b530c0 100644
--- a/omx/gstomx_base_filter.c
+++ b/omx/gstomx_base_filter.c
@@ -112,7 +112,12 @@ change_state (GstElement *element,
                 goto leave;
             }
             break;
-
+        case GST_STATE_CHANGE_READY_TO_PAUSED:
+            self->in_port->enabled = TRUE;
+            g_omx_port_resume (self->in_port);
+            self->out_port->enabled = TRUE;
+            g_omx_port_resume (self->out_port);
+            break;            
         default:
             break;
     }
-- 
1.7.1

