From 2bf91f72085187e3240bc3027b82cd064db169cf Mon Sep 17 00:00:00 2001
From: Olivier Naudan <o-naudan@ti.com>
Date: Mon, 18 Oct 2010 15:00:40 +0200
Subject: [PATCH] Fixed wrong shared lib names

---
 omx/gstomx.c |   30 +++++++++++++++---------------
 1 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 62f8074..2b00e85 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -71,24 +71,24 @@ typedef struct TableItem
 
 static TableItem element_table[] =
 {
-    { "omx_dummy",          "libOMX_Core.so",           "OMX.TI.DUCATI1.MISC.SAMPLE",   NULL,                   GST_RANK_NONE,      gst_omx_dummy_get_type },
-    { "omx_mpeg4dec",       "libOMX_Core.so",           "OMX.TI.DUCATI1.VIDEO.DECODER", "video_decoder.mpeg4",  GST_RANK_PRIMARY,   gst_omx_mpeg4dec_get_type },
-    { "omx_h264dec",        "libOMX_Core.so",           "OMX.TI.DUCATI1.VIDEO.DECODER", "video_decoder.avc",    GST_RANK_PRIMARY,   gst_omx_h264dec_get_type },
-    { "omx_h263dec",        "libOMX_Core.so",           "OMX.TI.DUCATI1.VIDEO.DECODER", "video_decoder.h263",   GST_RANK_PRIMARY,   gst_omx_h263dec_get_type },
-    { "omx_vp6dec",         "libOMX_Core.so",           "OMX.TI.DUCATI1.VIDEO.DECODER", "video_decoder.vp6",    GST_RANK_PRIMARY,   gst_omx_vp6dec_get_type },
-    { "omx_wmvdec",         "libOMX_Core.so",           "OMX.TI.Video.Decoder",         NULL,                   GST_RANK_NONE,      gst_omx_wmvdec_get_type },
-    { "omx_mpeg4enc",       "libOMX_Core.so",           "OMX.TI.DUCATI1.VIDEO.MPEG4E",  NULL,                   GST_RANK_PRIMARY,   gst_omx_mpeg4enc_get_type },
-    { "omx_h264enc",        "libOMX_Core.so",           "OMX.TI.DUCATI1.VIDEO.H264E",   NULL,                   GST_RANK_PRIMARY,   gst_omx_h264enc_get_type },
-    { "omx_h263enc",        "libOMX_Core.so",           "OMX.TI.DUCATI1.VIDEO.MPEG4E",  NULL,                   GST_RANK_PRIMARY,   gst_omx_h263enc_get_type },
+    { "omx_dummy",          "libOMX_Core.so.0",           "OMX.TI.DUCATI1.MISC.SAMPLE",   NULL,                   GST_RANK_NONE,      gst_omx_dummy_get_type },
+    { "omx_mpeg4dec",       "libOMX_Core.so.0",           "OMX.TI.DUCATI1.VIDEO.DECODER", "video_decoder.mpeg4",  GST_RANK_PRIMARY,   gst_omx_mpeg4dec_get_type },
+    { "omx_h264dec",        "libOMX_Core.so.0",           "OMX.TI.DUCATI1.VIDEO.DECODER", "video_decoder.avc",    GST_RANK_PRIMARY,   gst_omx_h264dec_get_type },
+    { "omx_h263dec",        "libOMX_Core.so.0",           "OMX.TI.DUCATI1.VIDEO.DECODER", "video_decoder.h263",   GST_RANK_PRIMARY,   gst_omx_h263dec_get_type },
+    { "omx_vp6dec",         "libOMX_Core.so.0",           "OMX.TI.DUCATI1.VIDEO.DECODER", "video_decoder.vp6",    GST_RANK_PRIMARY,   gst_omx_vp6dec_get_type },
+    { "omx_wmvdec",         "libOMX_Core.so.0",           "OMX.TI.Video.Decoder",         NULL,                   GST_RANK_NONE,      gst_omx_wmvdec_get_type },
+    { "omx_mpeg4enc",       "libOMX_Core.so.0",           "OMX.TI.DUCATI1.VIDEO.MPEG4E",  NULL,                   GST_RANK_PRIMARY,   gst_omx_mpeg4enc_get_type },
+    { "omx_h264enc",        "libOMX_Core.so.0",           "OMX.TI.DUCATI1.VIDEO.H264E",   NULL,                   GST_RANK_PRIMARY,   gst_omx_h264enc_get_type },
+    { "omx_h263enc",        "libOMX_Core.so.0",           "OMX.TI.DUCATI1.VIDEO.MPEG4E",  NULL,                   GST_RANK_PRIMARY,   gst_omx_h263enc_get_type },
     { "omx_vorbisdec",      "libomxil-bellagio.so.0",   "OMX.st.audio_decoder.ogg.single", NULL,                GST_RANK_NONE,   gst_omx_vorbisdec_get_type },
-    { "omx_mp3dec",         "libOMX_Core.so",           "OMX.TI.AUDIO.DECODE",          "audio_decode.dsp.mp3", GST_RANK_NONE,   gst_omx_mp3dec_get_type },
+    { "omx_mp3dec",         "libOMX_Core.so.0",           "OMX.TI.AUDIO.DECODE",          "audio_decode.dsp.mp3", GST_RANK_NONE,   gst_omx_mp3dec_get_type },
     { "omx_mp2dec",         "libomxil-bellagio.so.0",   "OMX.st.audio_decoder.mp3.mad", NULL,                   GST_RANK_NONE,   gst_omx_mp2dec_get_type },
     { "omx_amrnbdec",       "libomxil-bellagio.so.0",   "OMX.st.audio_decoder.amrnb",   NULL,                   GST_RANK_NONE,   gst_omx_amrnbdec_get_type },
     { "omx_amrnbenc",       "libomxil-bellagio.so.0",   "OMX.st.audio_encoder.amrnb",   NULL,                   GST_RANK_NONE,   gst_omx_amrnbenc_get_type },
     { "omx_amrwbdec",       "libomxil-bellagio.so.0",   "OMX.st.audio_decoder.amrwb",   NULL,                   GST_RANK_NONE,   gst_omx_amrwbdec_get_type },
     { "omx_amrwbenc",       "libomxil-bellagio.so.0",   "OMX.st.audio_encoder.amrwb",   NULL,                   GST_RANK_NONE,   gst_omx_amrwbenc_get_type },
-    { "omx_aacdec",         "libOMX_Core.so",           "OMX.TI.AUDIO.DECODE",          "audio_decode.dsp.aac", GST_RANK_NONE,   gst_omx_aacdec_get_type },
-    { "omx_aacenc",         "libOMX_Core.so",           "OMX.TI.AUDIO.ENCODE",          "audio_encode.dsp.aac", GST_RANK_NONE,   gst_omx_aacenc_get_type },
+    { "omx_aacdec",         "libOMX_Core.so.0",           "OMX.TI.AUDIO.DECODE",          "audio_decode.dsp.aac", GST_RANK_NONE,   gst_omx_aacdec_get_type },
+    { "omx_aacenc",         "libOMX_Core.so.0",           "OMX.TI.AUDIO.ENCODE",          "audio_encode.dsp.aac", GST_RANK_NONE,   gst_omx_aacenc_get_type },
     { "omx_adpcmdec",       "libomxil-bellagio.so.0",   "OMX.st.audio_decoder.adpcm",   NULL,                   GST_RANK_NONE,   gst_omx_adpcmdec_get_type },
     { "omx_adpcmenc",       "libomxil-bellagio.so.0",   "OMX.st.audio_encoder.adpcm",   NULL,                   GST_RANK_NONE,   gst_omx_adpcmenc_get_type },
     { "omx_g711dec",        "libomxil-bellagio.so.0",   "OMX.st.audio_decoder.g711",    NULL,                   GST_RANK_NONE,   gst_omx_g711dec_get_type },
@@ -97,13 +97,13 @@ static TableItem element_table[] =
     { "omx_g729enc",        "libomxil-bellagio.so.0",   "OMX.st.audio_encoder.g729",    NULL,                   GST_RANK_NONE,   gst_omx_g729enc_get_type },
     { "omx_ilbcdec",        "libomxil-bellagio.so.0",   "OMX.st.audio_decoder.ilbc",    NULL,                   GST_RANK_NONE,   gst_omx_ilbcdec_get_type },
     { "omx_ilbcenc",        "libomxil-bellagio.so.0",   "OMX.st.audio_encoder.ilbc",    NULL,                   GST_RANK_NONE,   gst_omx_ilbcenc_get_type },
-    { "omx_jpegenc",        "libOMX_Core.so",           "OMX.TI.JPEG.encoder",          NULL,                   GST_RANK_NONE,   gst_omx_jpegenc_get_type },
-    { "omx_jpegdec",        "libOMX_Core.so",           "OMX.TI.DUCATI1.IMAGE.JPEGD",   NULL,                   GST_RANK_NONE,   gst_omx_jpegdec_get_type },
+    { "omx_jpegenc",        "libOMX_Core.so.0",           "OMX.TI.JPEG.encoder",          NULL,                   GST_RANK_NONE,   gst_omx_jpegenc_get_type },
+    { "omx_jpegdec",        "libOMX_Core.so.0",           "OMX.TI.DUCATI1.IMAGE.JPEGD",   NULL,                   GST_RANK_NONE,   gst_omx_jpegdec_get_type },
     { "omx_audiosink",      "libomxil-bellagio.so.0",   "OMX.st.alsa.alsasink",         NULL,                   GST_RANK_NONE,      gst_omx_audiosink_get_type },
     { "omx_videosink",      "libomxil-bellagio.so.0",   "OMX.st.videosink",             NULL,                   GST_RANK_NONE,      gst_omx_videosink_get_type },
     { "omx_filereadersrc",  "libomxil-bellagio.so.0",   "OMX.st.audio_filereader",      NULL,                   GST_RANK_NONE,      gst_omx_filereadersrc_get_type },
     { "omx_volume",         "libomxil-bellagio.so.0",   "OMX.st.volume.component",      NULL,                   GST_RANK_NONE,      gst_omx_volume_get_type },
-    { "omx_camera",         "libOMX_Core.so",           "OMX.TI.DUCATI1.VIDEO.CAMERA",  NULL,                   GST_RANK_PRIMARY,   gst_omx_camera_get_type },
+    { "omx_camera",         "libOMX_Core.so.0",           "OMX.TI.DUCATI1.VIDEO.CAMERA",  NULL,                   GST_RANK_PRIMARY,   gst_omx_camera_get_type },
     { NULL, NULL, NULL, NULL, 0, NULL },
 };
 
-- 
1.7.1

