Index: tiopenmax-domx/system/omx_core/Makefile.am
===================================================================
--- tiopenmax-domx.orig/system/omx_core/Makefile.am	2010-11-04 17:01:13.000000000 +0100
+++ tiopenmax-domx/system/omx_core/Makefile.am	2010-11-04 17:07:41.000000000 +0100
@@ -11,7 +11,8 @@
 libOMX_Core_la_include_HEADERS = inc/OMX_ComponentRegistry.h inc/OMX_Debug.h inc/OMX_Image.h inc/OMX_Types.h \
 	inc/OMX_Audio.h inc/OMX_ContentPipe.h inc/OMX_DebugMem.h inc/OMX_Index.h inc/OMX_Video.h \
 	inc/OMX_Component.h inc/OMX_Core.h inc/OMX_IVCommon.h inc/OMX_Other.h inc/OMX_TI_Common.h \
-	inc/OMX_TI_IVCommon.h inc/OMX_TI_Index.h inc/OMX_TI_Video.h inc/OMX_TI_Core.h
+	inc/OMX_TI_IVCommon.h inc/OMX_TI_Index.h inc/OMX_TI_Video.h inc/OMX_TI_Core.h \
+        inc/OMX_ComponentExt.h inc/OMX_CoreExt.h inc/OMX_IndexExt.h inc/OMX_VideoExt.h
 libOMX_Core_la_includedir   = $(includedir)/omx
 
 pkgconfig_DATA          = libOMX_Core.pc
