diff -Nur -x '*.orig' -x '*~' tiopenmax-domx//system/omx_core/src/OMX_Core.c tiopenmax-domx.new//system/omx_core/src/OMX_Core.c
--- tiopenmax-domx//system/omx_core/src/OMX_Core.c	2010-10-04 17:22:11.000000000 +0200
+++ tiopenmax-domx.new//system/omx_core/src/OMX_Core.c	2010-10-04 17:26:29.000000000 +0200
@@ -166,7 +166,7 @@
     OMX_CALLBACKTYPE * pCallBacks)
 {
 	static const char prefix[] = "lib";
-	static const char postfix[] = ".so";
+	static const char postfix[] = ".so.0";
 	OMX_ERRORTYPE(*pComponentInit) (OMX_HANDLETYPE *);
 	OMX_ERRORTYPE eError = OMX_ErrorNone;
 	OMX_COMPONENTTYPE *componentType;
@@ -653,7 +653,7 @@
 	OMX_STRING temp = NULL;
 	static OMX_STRING namePrefix = "OMX";
 	static OMX_STRING filePrefix = "libOMX.";
-	static OMX_STRING suffix = ".so";
+	static OMX_STRING suffix = ".so.0";
 #endif
 	int j = 0;
 	int numFiles = 0;
