From 0224a8253f12720fbd0046c540dd97a0c0c7e597 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Thu, 23 Dec 2010 16:42:29 -0600
Subject: [PATCH 12/44] v4l2: fix typo

---
 sys/v4l2/gstv4l2bufferpool.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index b8c61ba..cf43909 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -263,7 +263,7 @@ gst_v4l2_buffer_pool_get_type (void)
 
   if (G_UNLIKELY (_gst_v4l2_buffer_pool_type == 0)) {
     static const GTypeInfo v4l2_buffer_pool_info = {
-      sizeof (GstBufferClass),
+      sizeof (GstMiniObjectClass),
       NULL,
       NULL,
       gst_v4l2_buffer_pool_class_init,
-- 
1.7.1

